### ISSUE TITLE
Docs: F811 (redefined-while-unused) silently skips names matching dummy-variable-rgx — every `_private` name by default

### ISSUE THREAD

**Issue by MoreAllLess** [astral-sh/ruff#27411]
## Summary

`F811` (`redefined-while-unused`) skips any name matching [`lint.dummy-variable-rgx`](https://docs.astral.sh/ruff/settings/#lint_dummy-variable-rgx). Because the default pattern is `^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$`, **every leading-underscore name is exempt from F811 by default** — functions and classes included, not just variables.

Neither page documents this:

- The [F811 rule page](https://docs.astral.sh/ruff/rules/redefined-while-unused/) doesn't mention `dummy-variable-rgx` at all, and lists no options.
- The [`dummy-variable-rgx` setting](https://docs.astral.sh/ruff/settings/#lint_dummy-variable-rgx) describes itself as applying to "unused-variable rules", which doesn't obviously cover a redefinition rule that fires on `def` and `class` statements.

This is a docs request, not a bug report — but see "Prior discussion" below, because a maintainer predicted this exact surprise two years ago.

## Reproduction

Identical bodies; only the name differs. [Playground](https://play.ruff.rs/)

```python
# flagged: F811 Redefinition of unused `f` from line 1
def f(): ...
def f(): ...
```

```python
# NOT flagged — "All checks passed!"
def _f(): ...
def _f(): ...
```

Confirmed on ruff 0.16.0 and 0.16.1, with `--isolated --no-cache --select F811`. Setting `lint.dummy-variable-rgx` to anything that doesn't match `_f` makes the second case fire immediately, which is what identifies the mechanism.

## Why this is worth documenting

We lost real time to this. A FastAPI router in our codebase contained a byte-for-byte duplicate `async def _send_web_push_new_ticket(...)` — defined twice, 24 lines apart, the first silently shadowed. `F` is in our `select`, CI runs `ruff check .` as a blocking gate, and it passed clean for months.

Because the rule *page* gives no hint that a name pattern is involved, triage went after every other plausible cause first: `# noqa` comments, `per-file-ignores`, a stale `.ruff_cache`, the "unused" precondition being satisfied by the call site. All dead ends. The mechanism is only findable by mutating the function *name*, which is not an obvious thing to try when the rule is about duplicate definitions.

The wider consequence is the part we'd flag for other users: this codebase names every private module helper `_helper()`, which is an extremely common Python convention. Under the default `dummy-variable-rgx`, **F811 had never applied to a single one of them.** The rule was effectively off for the entire private API surface of the project, with no signal that this was the case.

## Prior discussion

This isn't new information to the project — but it has never reached the docs.

@charliermarsh identified the same mechanism in #10522 on 2024-03-23, responding to a user who hit exactly this with private names, and noted it might not belong on functions, adding that users could find it "really surprising". That thread is a *rule request* ("Rule to detect duplicate function definitions"), so the finding is buried somewhere nobody investigating F811 behaviour would look, and nothing appears to have changed since.

#27095 is currently reconsidering the same pattern in a different rule — @MichaReiser's point there, that if the intent is excluding private members then the rule should do that explicitly "rather than depending on the dummy variable regex to accidentally catch them", applies equally here.

## Ask

Primarily documentation:

1. Note on the [F811 rule page](https://docs.astral.sh/ruff/rules/redefined-while-unused/) that the rule honours `dummy-variable-rgx`, and list it under Options as other rules do.
2. Broaden the `dummy-variable-rgx` description so it doesn't read as variable-only, or enumerate the rules that consult it.

Optionally, and following the direction of #27095: reconsider whether `dummy-variable-rgx` should apply to `def`/`class` bindings under F811 at all. A leading underscore signals "private", not "unused" — the two mean quite different things for a redefinition check, and the current coupling silently disables the rule across a naming convention that a large share of Python codebases use.

Happy to open a docs PR for (1) and (2) if that would be useful.

**Comment by ntBre** [comment]
This is already covered on the linked docs page: https://docs.astral.sh/ruff/rules/redefined-while-unused/#options

> [Options](https://docs.astral.sh/ruff/rules/redefined-while-unused/#options)
> This rule ignores dummy variables, as determined by:
> 
> - [lint.dummy-variable-rgx](https://docs.astral.sh/ruff/settings/#lint_dummy-variable-rgx)
